provisioning/service/src/Auth/AuthenticationMethodFactory.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Provisioning.Service
+{
+    using System;
+
+    /// <summary>
+    /// Creates an instance of an implementation of <see cref="IAuthenticationMethod"/> based on known authentication parameters.
+    /// </summary>
+    public sealed class AuthenticationMethodFactory
+    {
+        static internal IAuthenticationMethod GetAuthenticationMethod(ProvisioningConnectionStringBuilder iotHubConnectionStringBuilder)
```

Comment:
Nit: (here and everywhere) `internal static`

Reply:
Done